llvm/lib/ObjectYAML/GOFFEmitter.cpp

```diff
@@ -308,11 +308,14 @@ bool GOFFState::writeObject() {
   if (HasError)
     return false;
   // Iterate over all records.
-  for (const std::unique_ptr<llvm::GOFFYAML::RecordBase> &Rec : Doc.Records)
+  unsigned RecordNum = 0;
+  for (const std::unique_ptr<llvm::GOFFYAML::RecordBase> &Rec : Doc.Records) {
```

Comment:
You should look up the `enumerate` function to allow iterating over the range whilst also producing an index.